docs: clarify layout.js nesting behavior in route groups (#95720)
vercel/next.js: docs/01-app/01-getting-started/02-project-structure.mdx

  height="930"
/>

> Even though routes inside `(marketing)` and `(shop)` share the same URL hierarchy, you can create a different layout for each group by adding a `layout.js` file inside their folders. This `layout.js` nests inside the top-level `app/layout.js` like any other layout, it does not replace it. To remove the top-level layout entirely and give each group its own root layout instead, see [Creating multiple root layouts](#creating-multiple-root-layouts) below.

Reviewer: I'm not convinced this needs to be called out here.

A `layout.js` in a route group is still just a layout. The root layout is simply the first layout encountered for a route. If `app/layout.js` exists, it remains the root and the group layout nests under it. If it doesn't, the group layout becomes the root and must define `<html>` and `<body>`. That configuration is already covered in the "Creating multiple root layouts" section.

I'd lean toward keeping this section focused unless we've seen multiple users get tripped up by this.

That said, if we do want a small clarification here, perhaps something like:

```suggestion
Even though routes inside `(marketing)` and `(shop)` share the same URL hierarchy, you can create a different layout for each group by adding a `layout.js` file inside their folders. These layouts nest within the existing app layout.
```
